js/PulsarFormComponent.js

```diff
@@ -84,9 +84,24 @@ PulsarFormComponent.prototype.initTimePickers = function () {
  */
 PulsarFormComponent.prototype.initDatePickers = function () {
     // Attach basic pikaday to datepicker fields
-    this.$html
-        .find('[data-datepicker=true]')
-        .pikaday({ format: 'DD/MM/YYYY' });
+
+    const datepickers = this.$html.find('[data-datepicker="true"]');
+
+    datepickers.each((index, element) => {
+        const $element = $(element);
```

Comment:
I think we should cache the parent here instead of the element, as it looks as though it's the parent which is used as the base for the `.find(...)` query.

Something like...

```
const $parent = $(element).parent();

$parent.find(...).pikaday();
$parent.find(...).pikaday();
$parent.find(...).pikaday();
$parent.find(...).pikaday();
$parent.find(...).pikaday();
$parent.find(...).pikaday();
```

Also on this, is there a reason we find the datepicker's parent and then go back down into the datepicker. As we're controlling the markup here, can't we just check the value of `data-datepickerStyle` on the `element` in this context?

Reply:
The data-datepickerStyle and data-datepicker are within the same input

Reply:
That is why I ask. In the code for each `data-datepicker` element you find it's parent then query an attribute that exists on the `data-datepicker` element. When we would simple check the value of that attribute on that element directly...

```
things.forEach((index, element) => {
    // ?
    const format = $(element).attr('data-datePickerStyle');
});
```